### PR TITLE
Backtrace options

### PR DESCRIPTION
**Pull request by BriocheBerlin**
Closes https://github.com/knurling-rs/probe-run/issues/198

### CODE REVIEW

File: tests/test.rs [hunk 1]
```diff
@@ -75,17 +75,6 @@ fn successful_run_has_no_backtrace() {
     insta::assert_snapshot!(run_result.output);
 }
 
-#[test]
-// this test should not be run by default, as it requires the target hardware to be present
-#[ignore]
-fn successful_run_can_enforce_backtrace() {
```
Comment (Lotterleben) [inline tests/test.rs:81]: why is this deleted and not adapted?

File: src/backtrace/mod.rs [hunk 2]
```diff
@@ -57,10 +48,18 @@ pub(crate) fn print(
...
+    let print_backtrace= match settings.backtrace {
+        BacktraceOptions::Never => false,
+        BacktraceOptions::Always => true,
+        BacktraceOptions::Auto => if settings.panic_present
+            || unwind.outcome == Outcome::StackOverflow
+            || unwind.corrupted
+            || contains_exception {
+                true
+            } else {
+                false
```
Comment (Lotterleben) [inline src/backtrace/mod.rs:60]: this `if...else` can be left out because the `settings.panic_present || unwind.outcome == Outcome::StackOverflow || ...` statement itself evaluates to a `bool`, which is what we want to return. e.g.:

```rust
 BacktraceOptions::Auto => settings.panic_present
             || unwind.outcome == Outcome::StackOverflow
             || unwind.corrupted
             || contains_exception 
```
should work as well

File: src/backtrace/mod.rs [hunk 3]
```diff
@@ -57,10 +48,18 @@ pub(crate) fn print(
         .iter()
         .any(|raw_frame| raw_frame.is_exception());
 
-    let print_backtrace = settings.backtrace
-        || unwind.outcome == Outcome::StackOverflow
-        || unwind.corrupted
-        || contains_exception;
+    let print_backtrace= match settings.backtrace {
```
Comment (Lotterleben) [inline src/backtrace/mod.rs:51]: ha, neat!

File: tests/test.rs (hunk 1 above)
Comment (BriocheBerlin) [reply tests/test.rs:81]: Didn't realize at first that we keep the functionality (under a different name). Adapted!

File: src/backtrace/mod.rs [hunk 4]
```diff
@@ -8,10 +8,29 @@ mod pp;
...
 mod unwind;
 
+#[derive(PartialEq, Eq)]
+pub(crate) enum BacktraceOptions {
+    Auto,
+    Never,
+    Always,
+}
+
+impl From<&String> for BacktraceOptions {
```
[the remark this answers is not in the document]
Comment (BriocheBerlin) [reply src/backtrace/mod.rs:18]: Thanks, will leave (merge) for now but probably implement later.
